code/integration-tests/runtime-tests/tests/utils/multihopUtils.ts

```diff
@@ -0,0 +1,439 @@
+          asset.balance.set(address, new BigNumber(await queryNativeBalance(api, address)));
+        } else {
+          asset.balance.set(address, new BigNumber(await queryTokenBalance(api, address, asset.id)));
+        }
+      } else if (ecosystem === 'cosmos') {
+        if (asset.chain === 'centauri') {
+          asset.balance.set(address, new BigNumber(await queryBalanceOnCentauri(address, asset.id)));
+        } else {
+          asset.balance.set(address, new BigNumber(await queryBalanceOnOsmosis(address, asset.id)));
+        }
+      }
+    }))
+  }
+}
+
+export async function getTotalIssuance(asset: Asset | Asset[], ecosystem: string, api?: ApiPromise) {
+  if (Array.isArray(asset)) {
+    await Promise.all(asset.map(async specAsset => {
+      if (ecosystem === 'dotsama' && api !== undefined) {
+        if (specAsset.isNative) {
+          specAsset.totalIssuance = new BigNumber((await api.query.balances.totalIssuance()).toString());
+        } else {
+          specAsset.totalIssuance = new BigNumber((await api.query.tokens.totalIssuance(specAsset.id)).toString());
+        }
+      } else if (ecosystem === 'cosmos') {
+        if (specAsset.chain === 'centauri') {
+          specAsset.totalIssuance = new BigNumber(await queryTotalIssuanceOfTokenOnCentauri(specAsset.id));
+        } else {
+          specAsset.totalIssuance = new BigNumber(await queryTotalIssuanceOnOsmosis(specAsset.id));
+        }
+      }
+    }));
+  } else {
+    if (ecosystem === 'dotsama' && api !== undefined) {
+      if (asset.isNative) {
+        asset.totalIssuance = new BigNumber((await api.query.balances.totalIssuance()).toString());
+      } else {
+        asset.totalIssuance = new BigNumber((await api.query.tokens.totalIssuance(asset.id)).toString())
+      }
+    } else if (ecosystem === 'cosmos') {
+      if (asset.chain === 'centauri') {
+        asset.totalIssuance = new BigNumber(await queryTotalIssuanceOfTokenOnCentauri(asset.id));
+      } else {
+        asset.totalIssuance = new BigNumber(await queryTotalIssuanceOnOsmosis(asset.id));
+      }
+    }
+  }
+}
+
+export function mapRoutesAndChannels(startChain: string, endChain: string) {
+  const channels = returnChannels();
+  // @ts-ignore
+  const chann = channels.find(chains => {
+    if (chains.from === startChain && chains.to === endChain) {
+      return chains;
+    }
+  })
+  if (chann) return chann;
+  throw new Error('Channel not found');
+}
+
+function isSubstrate(startChain: string, endChain: string) {
+  if (startChain === 'polkadot' || endChain === 'polkadot') return false;
+  if (startChain === 'kusama' || endChain === 'kusama') return false;
+  if (endChain === 'centauri' || endChain === 'osmo') return false;
+  return true;
+}
+
+function isCosmos(startChain: string, endChain: string) {
+  if (endChain === 'centauri' || endChain === 'osmo') return true;
+  return false;
+}
+
+function returnChannels() {
+  return [
+    {from: 'picasso', to: 'composable', channelId: 1},
```

Comment:
please move to JSON this config

Reply:
I have cvm.json with such thing. may add picasso/composable here. to avoid copy paste

Reply:
moved to config.json file.